condition metrics(#188).ipynb

```diff
@@ -0,0 +1,1115 @@
+    "# regularly spaced binnings, disparately spaced binnings that correspond to\n",
+    "# sets of an equal amount of data points (such as quartiles, deciles, or\n",
+    "# n-quantiles), or any other arbitrary irregular binning can be used.\n",
+    "\n",
+    "# The function compute the accuracy score, recall score and f1 score for each partition and \n",
+    "# show them in bar diagrams\n",
+    "\n",
+    "\n",
+    "\n",
+    "def compute_quantiles(dataset, feature, quantiles=4):\n",
+    "    list_quantiles = np.arange(quantiles + 1) / quantiles\n",
+    "    edge_values = np.quantile(dataset[feature], list_quantiles)\n",
+    "    return edge_values\n",
+    "\n",
+    "def metrics_feature(\n",
+    "    test_dataset,\n",
+    "    test_predictions,\n",
+    "    feature,\n",
+    "    categorical=False,\n",
+    "    bins=10,\n",
+    "    bins_type=\"regular\",\n",
+    "    \n",
+    "):  \n",
+    "    if 'pred' not in test_dataset.columns:\n",
+    "        test_dataset['pred'] = test_predictions\n",
+    "        \n",
+    "    pred_col_index = test_dataset.columns.get_loc('pred')\n",
+    "    y_test_name = list(testset.columns)[pred_col_index-1]\n",
+    "    if categorical is False:\n",
+    "\n",
+    "        # Computes position of bin edges for quartiles or deciles\n",
+    "        if bins == \"quartiles\":\n",
+    "            lbls = ['1','2','3','4']\n",
+    "            bins = compute_quantiles(test_dataset, feature, quantiles=4)\n",
+    "            test_dataset['quantiles_'+feature] = pd.qcut(test_dataset[feature], q=4,duplicates=\"drop\",labels= lbls)\n",
+    "        elif bins == \"deciles\":\n",
+    "            lbls = ['1','2','3','4','5','6','7','8','9','10']\n",
+    "            bins = compute_quantiles(test_dataset, feature, quantiles=10)\n",
+    "            test_dataset['quantiles_'+feature] = pd.qcut(test_dataset[feature], q=10,duplicates=\"drop\",labels= lbls)\n",
+    "        elif type(bins) == int and bins_type == \"quantiles\":\n",
+    "            lbls = []\n",
+    "            bin_num = bins\n",
+    "            for i in range(1,bins+1):\n",
+    "                lbls.append(str(i))\n",
+    "            bins = compute_quantiles(test_dataset, feature, quantiles=bins)\n",
+    "            test_dataset['quantiles_'+feature] = pd.qcut(test_dataset[feature], q=bin_num,duplicates=\"drop\",labels= lbls)\n",
+    "        \n",
+    "        \n",
+    "        total_histogram_counts, bins = np.histogram(test_dataset[feature], bins)\n",
+    "        \n",
+    "        accuracy_score_list=[]\n",
+    "        recall_score_list = []\n",
+    "        f1_score_list = []\n",
+    "        \n",
+    "        \n",
+    "        ThisGroup = []\n",
+    "        \n",
+    "        for index in range(1,len(total_histogram_counts)+1):\n",
+    "                ThisGroup = test_dataset[test_dataset['quantiles_'+feature]== str(index)]\n",
```

Comment:
Since you are only using the `qcut` result to group the rows, you don't need to add it to the dataset and worry about naming the column - you can just keep it as a variable (eg. just `x = pd.qcut(...)` above). Since it's created based on a DataFrame column, it'll have the same index and grouping will work as expected.

Reply:
Got that. I will make changes to that.